server/controllers/authController.api.js

```diff
@@ -39,3 +50,27 @@ exports.login = (req, res) => {
         }
     });
 }
+
+exports.userById = (req, res, next, id) => {
+    User.findById(id).exec((err, user) => {
+        if (err || !user) {
+            return res.status(400).json({
+                error: 'User not found'
+            });
+        }
+        const { _id, name, email } = user;
+        req.profile = { _id, name, email };
+        next();
+    });
+}
+
+exports.isAuth = (req, res, next) => {
+    let user = req.profile && req.user._id && req.profile._id == req.user._id;
+    if (!user) {
+        return res.status(403).json({
+            // 403 means unauthorized access
```

Comment:
Not sure if this comment is really needed to be honest

Reply:
Sorry that was a note! lol